src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -137,7 +195,34 @@ protected override void InsertItem(int index, T item)
 
             OnCountPropertyChanged();
             OnIndexerPropertyChanged();
-            OnCollectionChanged(NotifyCollectionChangedAction.Add, item, index);
+
+            if (!_skipCollectionChanged)
+            {
+                OnCollectionChanged(NotifyCollectionChangedAction.Add, item, index);
+            }
+        }
+
+        /// <summary>
+        /// Called by base class Collection&lt;T&gt; when a collection of items is added to list;
+        /// raises a CollectionChanged event to any listeners.
+        /// </summary>
+        protected override void InsertItemsRange(int index, IEnumerable<T> collection)
+        {
+            CheckReentrancy();
+
+            _skipCollectionChanged = true;
+            try
+            {
+                base.InsertItemsRange(index, collection);
+            }
+            finally
+            {
+                _skipCollectionChanged = false;
+            }
+
+            OnCountPropertyChanged();
+            OnIndexerPropertyChanged();
+            OnCollectionChanged(NotifyCollectionChangedAction.Add, collection, index);
```

Comment:
Same comments here as https://github.com/dotnet/corefx/pull/35772/files#r263996280. This call doesn't appear to be correct, it's passing `collection` incorrectly as a single item that's changed. Instead it should be creating a `NotifyCollectionChangedEventArgs` using the constructor that allows specifying the multiple items that changed.

Like with my comment [here](https://github.com/dotnet/corefx/pull/35772/files#r263265676), unfortunately, the `NotifyCollectionChangedEventArgs` constructor only accepts an `IList` so if `collection` doesn't implement `IList`, it will need to be copied to an object that implements `IList` such as an array.

Reply:
I updated the `OnCollectionChanged` method to handle `IList` correctly. Let me know if there is anything else for this.